### ISSUE TITLE
Test failure in 32bit (t-nmod32 - vec_add)

### ISSUE THREAD

**Issue by tornaria** [flintlib/flint#1546]
```
$ build/gr/test/t-nmod32 
nmod32....vec_add
Integers mod 4026531840 (nmod32)
aliasing: 2
[268435522]
[66]

FAIL
Aborted
```

FWIW, `268435522 = 2^28 + 66` and `4026531840 = 15 * 2^28 = 2^32 - 2^28`.

**Comment by tornaria** [comment]
It seems to me that this function from `src/nmod.h` is bad:
```c
NMOD_INLINE
mp_limb_t _nmod_add(mp_limb_t a, mp_limb_t b, nmod_t mod)
{
   const mp_limb_t sum = a + b;
   return sum - mod.n + ((((mp_limb_signed_t)(sum - mod.n))>>(FLINT_BITS - 1)) & mod.n);
}
```

In fact,
```c
        nmod_t mod;
        mp_limb_t a = 0, b=0, ab;
        for(mod.n=2147483646U; mod.n<2147483652U; mod.n++) {
                ab = _nmod_add(a, b, mod);
                flint_printf("_nmod_add(%u, %u, %u) = %u\n", a, b, mod.n, ab);
        }
```
outputs
```
_nmod_add(0, 0, 2147483646) = 0
_nmod_add(0, 0, 2147483647) = 0
_nmod_add(0, 0, 2147483648) = 0
_nmod_add(0, 0, 2147483649) = 2147483647
_nmod_add(0, 0, 2147483650) = 2147483646
_nmod_add(0, 0, 2147483651) = 2147483645
```
For the record, `2147483648 == 2^31`; every modulus larger than that is broken in 32 bits.

I wonder if mod larger than 2^63 will also be broken in 64 bits, or there is something to avoid that.

**Comment by fredrik-johansson** [comment]
``_nmod_add`` is documented as working up to ``FLINT_BITS - 1`` bits which is by design. So it works for all 32-bit moduli on a 64-bit machine but on 32-bit machines ``nmod32`` will have to use ``nmod_add`` instead. Will fix; thanks for reporting!

**Comment by fredrik-johansson** [comment]
Should be fixed in 7d046723b1d74e66605c111483756322ebb1bc8d. Testing would be appreciated. Unfortunately we don't have 32-bit CI.

**Comment by tornaria** [comment]
> Should be fixed in [7d04672](https://github.com/flintlib/flint/commit/7d046723b1d74e66605c111483756322ebb1bc8d). Testing would be appreciated. Unfortunately we don't have 32-bit CI.

Works fine, thanks!

**Comment by fredrik-johansson** [comment]
Excellent! This will be in 3.0.1.
